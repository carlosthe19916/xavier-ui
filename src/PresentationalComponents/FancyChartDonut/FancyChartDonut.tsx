import React, { Component } from 'react';
import {
    ChartDonut,
    ChartDonutProps,
    ChartLegend,
    ChartLegendProps
} from '@patternfly/react-charts';

export interface FancyChartDonutData {
    label: string;
    value: number;
    color: string;
}

interface Props {
    data: FancyChartDonutData[];
    chartProps?: ChartDonutProps;
    chartLegendProps?: ChartLegendProps;
    tickFormat?: (label: string, value: number) => string;
}

interface State {
}

class FancyChartDonut extends Component<Props, State> {

    constructor(props: Props) {
        super(props);
    }

    render() {
        const { data, chartProps, chartLegendProps, tickFormat } = this.props;

        const chartData = data.map((val) => {
            return {
                x: val.label,
                y: val.value
            };
        });

        const legendData = data.map((val) => {
            return {
                name: tickFormat ? tickFormat (val.label, val.value) : `${val.label}: ${val.value}`
            };
        });

        const colorScale = data.map((val) => val.color);

        return (
            <React.Fragment>
                <div className="pf-u-display-flex">
                    <div className="donut-chart-container">
                        <ChartDonut
                            data={ chartData }
                            colorScale={ colorScale }
                            labels={ datum => tickFormat ? tickFormat(datum.x, datum.y) : `${datum.x}: ${datum.y}` }
                            { ...chartProps }
                        />
                    </div>
                    <ChartLegend
                        data={ legendData }
                        colorScale={ colorScale }
                        orientation="vertical"
                        { ...chartLegendProps }
                    />
                </div>
            </React.Fragment>
        );
    }
}

export default FancyChartDonut;
