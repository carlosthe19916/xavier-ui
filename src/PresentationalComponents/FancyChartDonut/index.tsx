import FancyChartDonut from './FancyChartDonut';

export default FancyChartDonut;
