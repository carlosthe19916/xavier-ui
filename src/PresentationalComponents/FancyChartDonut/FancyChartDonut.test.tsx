import React from "react";
import { shallow } from "enzyme";
import FancyChartDonut, { FancyChartDonutData } from "./FancyChartDonut";

const chartData: FancyChartDonutData[] = [
    { label: 'year1', value: 10, color: 'red', data: {} },
    { label: 'year2', value: 20, color: 'blue', data: {} },
    { label: 'year3', value: 30, color: 'yellow', data: {} },
    { label: 'year4', value: 40, color: 'green', data: {} }
];

describe("FancyChartDonut", () => {
  
  it("expect to render with minimun props", () => {
    const data = [
      ...chartData
    ];
    const wrapper = shallow(<FancyChartDonut data={ data } />);
    expect(wrapper).toMatchSnapshot();
  });

//   it("expect to render footer", () => {
//     const data = {
//         ...baseData
//     };
    
//     const footer = (
//         <p>this is a footer</p>
//     );

//     const wrapper = shallow(<FancyGroupedBarChart data={ data } footer={ footer } />);
//     expect(wrapper).toMatchSnapshot();
//   });
  
});
