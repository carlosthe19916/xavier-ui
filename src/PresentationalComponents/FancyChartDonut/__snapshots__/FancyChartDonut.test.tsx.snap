// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FancyChartDonut expect to render with minimun props 1`] = `
<Fragment>
  <div
    className="pf-u-display-flex"
  >
    <div
      className="donut-chart-container"
    >
      <ChartDonut
        colorScale={
          Array [
            "red",
            "blue",
            "yellow",
            "green",
          ]
        }
        data={
          Array [
            Object {
              "data": Object {},
              "x": "year1",
              "y": 10,
            },
            Object {
              "data": Object {},
              "x": "year2",
              "y": 20,
            },
            Object {
              "data": Object {},
              "x": "year3",
              "y": 30,
            },
            Object {
              "data": Object {},
              "x": "year4",
              "y": 40,
            },
          ]
        }
        labelComponent={<ChartTooltip />}
        labels={[Function]}
      />
    </div>
    <ChartLegend
      colorScale={
        Array [
          "red",
          "blue",
          "yellow",
          "green",
        ]
      }
      data={
        Array [
          Object {
            "name": "year1: 10",
          },
          Object {
            "name": "year2: 20",
          },
          Object {
            "name": "year3: 30",
          },
          Object {
            "name": "year4: 40",
          },
        ]
      }
      orientation="vertical"
    />
  </div>
</Fragment>
`;
